pkg/datarepair/checker/checker.go

```diff
@@ -85,9 +87,21 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 	var remoteSegmentsLost int64
 	var remoteSegmentInfo []string
 
-	err = checker.metainfo.Iterate("", "", true, false,
+	err = checker.metainfo.Iterate("", checker.lastChecked, true, false,
 		func(it storage.Iterator) error {
 			var item storage.ListItem
+
+			defer func() {
```

Comment:
I think we could log errors instead of returning them. But I don't know what the best option would be. The way we are currently implementing it (returning errors from the loop) seems consistent with the logic we are using in the audit service (`pkg/audit/service.go`)

Reply:
In pkg/discovery/service.go which also uses the Cycle loop, it looks like `discovery.Run` calls `discovery.Refresh.Start`, `discovery.Graveyard.Start`, and `discovery.Discovery.Start` - those will return errors and then the service will log the errors. Seems like a similar deal in storagenode/orders/sender.go for the sender.Run call, which logs errors.